src/overlays/actors/ovl_En_Ossan/z_en_ossan.c

```diff
@@ -37,227 +114,2406 @@ const ActorInit En_Ossan_InitVars = {
+        }
+    }
+}
+
+void EnOssan_State_LookFromShelfToShopkeeper(EnOssan* this, GlobalContext* globalCtx, Player* player) {
+    Math_ApproachF(&this->cameraFaceAngle, 0.0f, 0.5f, 10.0f);
+    if ((this->cameraFaceAngle < 0.5f) && (this->cameraFaceAngle > -0.5f)) {
+        EnOssan_UpdateCameraDirection(this, globalCtx, 0.0f);
+    }
+    EnOssan_UpdateCameraDirection(this, globalCtx, this->cameraFaceAngle);
+    if (this->cameraFaceAngle == 0.0f) {
+        EnOssan_StartShopping(globalCtx, this);
+    }
+}
+
+void EnOssan_State_DisplayOnlyBombDialog(EnOssan* this, GlobalContext* globalCtx, Player* player) {
+    if (!EnOssan_ReturnItemToShelf(this)) {
+        osSyncPrintf("%s[%d]:\x1b[32mズーム中！！\x1b[m\n", "../z_en_oB1.c", 2355);
+        return;
+    }
+    Math_ApproachF(&this->cameraFaceAngle, 0.0f, 0.5f, 10.0f);
+    if (this->cameraFaceAngle < 0.5f && this->cameraFaceAngle > -0.5f) {
+        EnOssan_UpdateCameraDirection(this, globalCtx, 0.0f);
+    }
+    EnOssan_UpdateCameraDirection(this, globalCtx, this->cameraFaceAngle);
+    if (this->cameraFaceAngle == 0.0f) {
+        func_8010B720(globalCtx, 0x3010);
+        this->stateFlag = ENOSSAN_STATE_WAIT_FOR_DISPLAY_ONLY_BOMB_DIALOG;
+    }
+}
+
+void EnOssan_GiveItemWithFanfare(GlobalContext* globalCtx, EnOssan* this) {
+    Player* player = PLAYER;
+
+    osSyncPrintf("\n\x1b[33m初めて手にいれた！！\x1b[m\n\n");
+    func_8002F434(&this->actor, globalCtx, this->shelfSlots[this->cursorIndex]->getItemId, 120.0f, 120.0f);
+    globalCtx->msgCtx.msgMode = 0x36;
+    globalCtx->msgCtx.unk_E3E7 = 4;
+    player->stateFlags2 &= ~0x20000000;
+    func_800BC490(globalCtx, 1);
+    Interface_ChangeAlpha(50);
+    this->drawCursor = 0;
+    EnOssan_UpdateCameraDirection(this, globalCtx, 0.0f);
+    this->stateFlag = ENOSSAN_STATE_GIVE_ITEM_FANFARE;
+    osSyncPrintf("\x1b[33m持ち上げ開始！！\x1b[m\n\n");
+}
+
+void EnOssan_SetStateCantGetItem(GlobalContext* globalCtx, EnOssan* this, u16 textId) {
+    func_8010B720(globalCtx, textId);
+    this->stateFlag = ENOSSAN_STATE_CANT_GET_ITEM;
+}
+
+void EnOssan_SetStateQuickBuyDialog(GlobalContext* globalCtx, EnOssan* this, u16 textId) {
+    func_8010B720(globalCtx, textId);
+    this->stateFlag = ENOSSAN_STATE_QUICK_BUY;
+}
+
+void EnOssan_HandleCanBuyItem(GlobalContext* globalCtx, EnOssan* this) {
+    EnGirlA* selectedItem = this->shelfSlots[this->cursorIndex];
+
+    switch (selectedItem->canBuyFunc(globalCtx, selectedItem)) {
+        case CANBUY_RESULT_SUCCESS_FANFARE:
+            if (selectedItem->actor.params == SI_HYLIAN_SHIELD && gSaveContext.infTable[7] & 0x40) {
+                EnOssan_SetStateGiveDiscountDialog(globalCtx, this);
+                break;
+            } else {
+                EnOssan_GiveItemWithFanfare(globalCtx, this);
+                this->drawCursor = 0;
+                this->shopItemSelectedTween = 0.0f;
+                selectedItem->setOutOfStockFunc(globalCtx, selectedItem);
+                return;
```

Comment:
These returns should probably all be breaks, also I think there should be a single break in this switch case